client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -213,18 +217,30 @@ private void submitRetryPushData(
           batchId,
           newLoc);
       try {
-        if (!testRetryRevive || remainReviveTimes < 1) {
-          TransportClient client =
-              dataClientFactory.createClient(newLoc.getHost(), newLoc.getPushPort(), partitionId);
-          NettyManagedBuffer newBuffer = new NettyManagedBuffer(Unpooled.wrappedBuffer(body));
-          String shuffleKey = Utils.makeShuffleKey(applicationId, shuffleId);
-
-          PushData newPushData =
-              new PushData(MASTER_MODE, shuffleKey, newLoc.getUniqueId(), newBuffer);
-          client.pushData(newPushData, pushDataTimeout, wrappedCallback);
+        // If shuffleClientBlacklistEnabled = false, blacklist should be empty.
+        if (blacklist.contains(newLoc.hostAndPushPort())) {
```

Comment:
Better to extract L221-L227 to a common method like checkBlacklist()

Reply:
How about current?